creusot-contracts/src/std/slice.rs

```diff
@@ -365,6 +365,18 @@ extern_spec! {
         #[ensures(ix.has_value(self@, *result))]
         fn index(&self, ix: I) -> &<[T] as Index<I>>::Output;
     }
+
+    mod std {
+        mod slice {
+            #[pure]
+            #[ensures(result@.len() == 1 && result@[0] == *s)]
+            fn from_ref<T>(s: &T) -> &[T];
+
+            #[pure]
+            #[ensures(result@.len() == 1 && (*result)@[0] == *s && (^result)@[0] == ^s)]
```

Comment:
```suggestion
            #[ensures(result@.len() == 1)]
            #[ensures(result@[0] == *s)]
            #[ensures((^result)@.len() == 1)]
            #[ensures((^result)@[0] == ^s)]
```

We need to ensure that the length of the prophecy is 1, too.

Some day we will figure out a way to ensure freely that the length of the current and final values are equal.